AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1074,19 +1074,58 @@ - (void)__layout
   ASDisplayNodeAssertMainThread();
   ASDN::MutexLocker l(_propertyLock);
   CGRect bounds = self.bounds;
-  if (CGRectEqualToRect(bounds, CGRectZero)) {
-    // Performing layout on a zero-bounds view often results in frame calculations
-    // with negative sizes after applying margins, which will cause
-    // measureWithSizeRange: on subnodes to assert.
+
+  [self measureNodeWithBoundsIfNecessary:bounds];
+
+  // Performing layout on a zero-bounds view often results in frame calculations
+  // with negative sizes after applying margins, which will cause
+  // measureWithSizeRange: on subnodes to assert.
+  if (!CGRectEqualToRect(bounds, CGRectZero)) {
```

Comment:
This is probably the right trade-off, but unfortunately it can result in some side effects. Specifically, an element that does not enable clipping would normally be able to show contents it holds even if it is zero sized, but this depends on the layout pass occurring.

This condition should improve efficiency (a bit, in some cases) and is probably best to keep, but we should be aware of this trade-off and consider if it is necessary to document somewhere